FEATURE_PROCESS.adoc

```diff
@@ -0,0 +1,117 @@
+= WildFly Feature Development Process
+
+WildFly features can have one of four different maturity levels, “Experimental”, “Preview”, “Community” and an unnamed default level. Promotion from one level to another, or initial feature incorporation at a given level, requires meeting various standards in the basic areas of requirements analysis, implementation, testing and documentation. The following table outlines the various standards for each of the maturity levels.
+
+[cols="5"]
+|===
+|
+|Experimental
+|Preview
+|Community
+|Default
+|Feature Team
+|Component lead or other SME
+|Experimental plus: +
+
+3rd party with a different perspective, able to question the feature requirements and API
+|Preview plus:
+|Community plus:
+|Requirement Analysis
+|Understandable JIRA description with an orientation toward what/why and not just how
+|Approved WildFly Proposals document
+
+Nice-to-have requirements allowed.
+|Approved WildFly Proposals document
+
+Nice-to-have requirements have been converted to non-requirements or are moved to a future work section.
+|Approved WildFly Proposals document
+
+Nice-to-have requirements have been converted to non-requirements or are moved to a future work section
+|Implementation
+|Primary use cases covered.
+
+Code style standards followed.
+
+Management API has experimental metadata
+
+Feature not used at runtime if not in experimental level
+
+New libraries not provisioned if not in appropriate stability level ???
+
+Third party libraries in Final version??
+|All hard requirements in analysis covered
+
+Management API has preview metadata
+
+Feature not used at runtime if not in preview level
+
+New libraries not provisioned if not in appropriate stability level ???
+|Stable API and behavior.
+
+All hard requirements in analysis covered
+
+Management API has community metadata
+
+Feature not used at runtime if not in community level
+
+New libraries not provisioned if not in appropriate stability level ???
+|Stable API and behavior
+
+All hard requirements in analysis covered
+|Domain Transformation
+|Encouraged
+|Encouraged
+|Encouraged
+|Required
+|Component Validation
+|Acceptable Open Source License
```

Comment:
do we have somewhere a list of such license?

Reply:
Not that I'm aware of. Perhaps @bstansberry knows?